internal/rsyncd/rsyncd.go

```diff
@@ -24,24 +24,33 @@ type sendTransfer struct {
 	lastMatch int64
 }
 
+func NewServer(modules ...config.Module) *Server {
+	moduleMap := map[string]config.Module{}
+	for _, module := range modules {
+		moduleMap[module.Name] = module
+	}
+
+	return &Server{moduleMap}
+}
+
 type Server struct {
-	Modules map[string]config.Module
+	modules map[string]config.Module
 }
 
 func (s *Server) getModule(requestedModule string) (config.Module, error) {
-	m, ok := s.Modules[requestedModule]
+	m, ok := s.modules[requestedModule]
 	if !ok {
 		return config.Module{}, fmt.Errorf("no such module")
 	}
 	return m, nil
 }
 
 func (s *Server) formatModuleList() string {
```

Comment:
Actually, the format list would be good to have in deterministic order as well, i.e. using the slice at this level, too.

Reply:
Sure, done.